<?xml version="1.0"?>
<launch>
    <!-- LAUNCH INTERFACE -->
    <arg name="hardware_interface" default="PositionJointInterface"/>
    <arg name="robot_name" default="iiwa"/>
    <arg name="model" default="iiwa7"/>

    <!-- need to load joint names  -->
    <rosparam command="load" file="$(find iiwa_fri_ros)/config/joint_names.yaml" />

    <arg name="controllers" default="joint_state_controller PositionJointInterface_trajectory_controller"/>


    <include file="$(find iiwa_fri_description)/launch/$(arg model)_upload.launch">
            <arg name="hardware_interface" value="$(arg hardware_interface)"/>
            <arg name="robot_name" value="$(arg robot_name)" />
    </include>

    <!-- Hardware interface to be loaded /-->
    <param name="interface" value="$(arg hardware_interface)"/>

    <rosparam file="$(find iiwa_fri_ros)/config/$(arg robot_name)_fri_control.yaml" command="load" />

    <!-- Loads the controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
          output="screen" args="$(arg controllers) --shutdown-timeout 2" />

    <param name="hostname" type="string" value="192.170.10.2"/>
    <param name="port" value="30200"/>

    <!-- the real hardware interface /-->
    <node name="iiwa_fri_hardware" pkg="iiwa_fri_ros" type="IIWA_FRI_Controller" respawn="false" output="screen"/>

    <!-- Converts joint states to TF transforms for rviz, etc -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
          respawn="false" output="screen">
        <remap from="joint_states" to="/$(arg robot_name)/joint_states" />
    </node>

</launch>